### PR TITLE
Add RPC for Boot Nodes

### PR DESCRIPTION
**Pull request by GheisMohammadi**
## Description
This PR enhances the visibility and monitoring capabilities of boot nodes by adding an RPC server. Previously, boot nodes lacked any RPCs and did not expose any meta data, limiting our ability to monitor them effectively. By adding RPC for boot nodes, this PR introduces several APIs that allow for better and more efficient monitoring, enabling us to observe and track the connectivity and peers connected to the boot node.
 
The PR would add these RPC methods for both `hmyboot` and `hmybootv2` domains:
```shell
echo "=======[hmyboot_getNodeMetadata]================="

curl -d '{
  "jsonrpc":"2.0",
  "method":"hmyboot_getNodeMetadata",
  "params":[],
  "id":1
 }' -H 'Content-Type:application/json' -X POST '0.0.0.0:8888'

 #{"jsonrpc":"2.0","id":1,"result":{"node-unix-start-time":1724318966,"p2p-connectivity":{"connected":24,"not-connected":1,"total-known-peers":25},"peerid":"", "network":"localnet", "version":"Harmony (C) 2023. bootnode, version v8421-v2024.0.0-83-g1cb52f7f0-dirty (root@ 2024-08-22T17:29:12+0800)"}}

 echo "=======[hmyboot_getPeerInfo]================="

curl -d '{
  "jsonrpc":"2.0",
  "method":"hmyboot_getPeerInfo",
  "params":[],
  "id":1
 }' -H 'Content-Type:application/json' -X POST '0.0.0.0:8888'

 #{"jsonrpc":"2.0","id":1,"result":{"peerid":"","known-peers":[],"connected-peers":[],"C":{}}}
 
 echo "=======[hmyboot_protocolVersion]================="

curl -d '{
  "jsonrpc":"2.0",
  "method":"hmyboot_protocolVersion",
  "params":[],
  "id":1
 }' -H 'Content-Type:application/json' -X POST '0.0.0.0:8888'

 #{"jsonrpc":"2.0","id":1,"result":"0x1"}
```

Additionally, this PR includes a redefinition of the node and RPC folder structure, making the codebase more organized and well structured to add boot nodes functions as well. 

Importantly, **_the changes made in this PR do not alter any logic for Harmony nodes_**, ensuring that the merge is safe and will not affect existing functionality. This improvement aims to facilitate better network management and monitoring of boot nodes.

### CODE REVIEW

**Comment by Frozen** [comment]
Pls fix conflicts

**Comment by sophoah** [comment]
We don't need to update the PR and this is just a remark :
```golang
	httpPort := flag.Int("rpc_http_port", 9500, "port of the rpc http")
	wsPort := flag.Int("rpc_ws_port", 9800, "port of the rpc ws")
```
those default port are the same as the harmony node. We'll have to make sure node running boot and harmony node have distinct port cc @mur-me

File: cmd/bootnode/main.go [hunk 1]
```diff
@@ -88,7 +92,6 @@ var (
 
 func printVersion(me string) {
 	fmt.Fprintf(os.Stderr, "Harmony (C) 2019. %v, version %v-%v (%v %v)\n", path.Base(me), version, commit, builtBy, builtAt)
```
Comment (sophoah) [inline cmd/bootnode/main.go:94]: update 2019 to 2024

**Comment by mur-me** [comment]
> We don't need to update the PR and this is just a remark :
> 
> ```go
> 	httpPort := flag.Int("rpc_http_port", 9500, "port of the rpc http")
> 	wsPort := flag.Int("rpc_ws_port", 9800, "port of the rpc ws")
> ```
> 
> those default port are the same as the harmony node. We'll have to make sure node running boot and harmony node have distinct port cc @mur-me

Agreed, here could be a conflict if run it on one machine together, but this isn't a huge problem

**Review by Frozen — APPROVED** [review]

**Comment by GheisMohammadi** [comment]
> We don't need to update the PR and this is just a remark :
> 
> ```go
> 	httpPort := flag.Int("rpc_http_port", 9500, "port of the rpc http")
> 	wsPort := flag.Int("rpc_ws_port", 9800, "port of the rpc ws")
> ```
> 
> those default port are the same as the harmony node. We'll have to make sure node running boot and harmony node have distinct port cc @mur-me

I noticed this when I was adjusting the ports. I intentionally chose to standardize and unify the RPC ports across nodes. This way, users don't need to seek for a new port every time they want to access Boot RPCs—keeping it consistent with the Harmony node makes it more user-friendly. I think in terms of safety having less ports open would make more sense.
Additionally, I included a flag that allows us to change the Boot node RPC ports if both the boot and Harmony nodes are running on the same server, ensuring flexibility. I've already updated the localnet scripts to prevent any conflicts there.

That said, I'm open to changing the ports if we think it's necessary. Let me know your thoughts!

File: cmd/bootnode/main.go (hunk 1 above)
Comment (GheisMohammadi) [reply cmd/bootnode/main.go]: done

File: internal/configs/bootnode/bootnode.go [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+	P2P        P2pConfig
+	HTTP       HttpConfig
+	WS         WsConfig
+	RPCOpt     RpcOptConfig
+	BLSKeys    BlsConfig
+	Pprof      PprofConfig
+	Log        LogConfig
+	Sys        *SysConfig        `toml:",omitempty"`
+	Devnet     *DevnetConfig     `toml:",omitempty"`
+	Revert     *RevertConfig     `toml:",omitempty"`
```
Comment (sophoah) [inline internal/configs/bootnode/bootnode.go:29]: What's the use case for a revert function on the bootnode ?

Comment (sophoah) [reply internal/configs/bootnode/bootnode.go:29]: same question for BlsConfig, ShardDataConfig, CacheConfig, LegacyConfig, DevnetConfig, SysConfig 

in cmd/bootnode/main.go there is only a couple of new flag added.

File: internal/configs/bootnode/bootnode.go [hunk 3]
```diff
@@ -0,0 +1,397 @@
...
+	evmCallTimeout, err := time.ParseDuration(bnc.RPCOpt.EvmCallTimeout)
+	if err != nil {
+		evmCallTimeout, _ = time.ParseDuration(nodeconfig.DefaultEvmCallTimeout)
+		utils.Logger().Warn().
+			Str("provided", bnc.RPCOpt.EvmCallTimeout).
+			Dur("updated", evmCallTimeout).
+			Msg("Sanitizing invalid evm_call timeout")
+	}
+	return nodeconfig.RPCServerConfig{
+		HTTPEnabled:        bnc.HTTP.Enabled,
```
Comment (sophoah) [inline internal/configs/bootnode/bootnode.go:70]: for RPC, we don't need to enable the full list of RPC below right ?

File: internal/configs/bootnode/config.go [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+
+func (n NetworkType) String() string {
+	if n == "" {
+		return Testnet // default to testnet
+	}
+	return string(n)
+}
+
+// Global is the index of the global node configuration
+const (
```
Comment (sophoah) [inline internal/configs/bootnode/config.go:84]: bootnode doesn't care of the number of shards here. So let's remove that part

File: internal/configs/bootnode/bootnode.go (hunk 2 above)
Comment (GheisMohammadi) [reply internal/configs/bootnode/bootnode.go]: all are removed

File: internal/configs/bootnode/bootnode.go (hunk 3 above)
Comment (GheisMohammadi) [reply internal/configs/bootnode/bootnode.go:66]: These are given from Harmony config.

File: internal/configs/bootnode/config.go (hunk 4 above)
Comment (GheisMohammadi) [reply internal/configs/bootnode/config.go:82]: removed

File: internal/configs/bootnode/bootnode.go [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+	TrieNodeLimit   int           // Memory limit (MB) at which to flush the current in-memory trie to disk
+	TrieTimeLimit   time.Duration // Time limit after which to flush the current in-memory trie to disk
+	TriesInMemory   uint64        // Block number from the head stored in disk before exiting
+	Preimages       bool          // Whether to store preimage of trie key to the disk
+	SnapshotLimit   int           // Memory allowance (MB) to use for caching snapshot entries in memory
+	SnapshotNoBuild bool          // Whether the background generation is allowed
+	SnapshotWait    bool          // Wait for snapshot construction on startup
+}
+
+type PreimageConfig struct {
```
Comment (sophoah) [inline internal/configs/bootnode/bootnode.go:280]: do we need that for bootnode config ?

File: internal/configs/bootnode/bootnode.go [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+}
+
+type DevnetConfig struct {
+	NumShards   int
+	ShardSize   int
+	HmyNodeSize int
+	SlotsLimit  int // HIP-16: The absolute number of maximum effective slots per shard limit for each validator. 0 means no limit.
+}
+
+type CacheConfig struct {
```
Comment (sophoah) [inline internal/configs/bootnode/bootnode.go:269]: do we need that for bootnode config ?

File: internal/configs/bootnode/bootnode.go [hunk 7]
```diff
@@ -0,0 +1,316 @@
...
+	StakingRPCsEnabled bool   // Expose Staking RPCs
+	LegacyRPCsEnabled  bool   // Expose Legacy RPCs
+	RpcFilterFile      string // Define filters to enable/disable RPC exposure
+	RateLimterEnabled  bool   // Enable Rate limiter for RPC
+	RequestsPerSecond  int    // for RPC rate limiter
+	EvmCallTimeout     string // Timeout for eth_call
+	PreimagesEnabled   bool   // Expose preimage API
+}
+
+type DevnetConfig struct {
```
Comment (sophoah) [inline internal/configs/bootnode/bootnode.go:262]: do we need that for bootnode config ?

File: internal/configs/bootnode/bootnode.go [hunk 8]
```diff
@@ -0,0 +1,316 @@
...
+type LogContext struct {
+	IP   string
+	Port int
+}
+
+type SysConfig struct {
+	NtpServer string
+}
+
+type HttpConfig struct {
```
Comment (sophoah) [inline internal/configs/bootnode/bootnode.go:231]: there is no rosetta for bootnode right ?

File: internal/configs/bootnode/bootnode.go [hunk 9]
```diff
@@ -0,0 +1,316 @@
...
+
+type ShardDataConfig struct {
+	EnableShardData bool
+	DiskCount       int
+	ShardCount      int
+	CacheTime       int
+	CacheSize       int
+}
+
+type GasPriceOracleConfig struct {
```
Comment (sophoah) [inline internal/configs/bootnode/bootnode.go:164]: do we need that for bootnode config ?

File: internal/configs/bootnode/bootnode.go [hunk 10]
```diff
@@ -0,0 +1,316 @@
...
+
+	PDAddr                      []string
+	Role                        string
+	StateDBCacheSizeInMB        uint32
+	StateDBCachePersistencePath string
+	StateDBRedisServerAddr      []string
+	StateDBRedisLRUTimeInDay    uint32
+}
+
+type ShardDataConfig struct {
```
Comment (sophoah) [inline internal/configs/bootnode/bootnode.go:156]: do we need that for bootnode config ?

File: internal/configs/bootnode/bootnode.go [hunk 11]
```diff
@@ -0,0 +1,316 @@
...
+	IsBackup               bool
+	IsBeaconArchival       bool
+	IsOffline              bool
+	DataDir                string
+	TraceEnable            bool
+	EnablePruneBeaconChain bool
+	RunElasticMode         bool
+}
+
+type TiKVConfig struct {
```
Comment (sophoah) [inline internal/configs/bootnode/bootnode.go:145]: do we need that for bootnode config ?

File: internal/configs/bootnode/bootnode.go [hunk 12]
```diff
@@ -0,0 +1,316 @@
...
+	UserAgent string
+	// p2p dial timeout
+	DialTimeout time.Duration
+	// P2P multiplexer type, should be comma separated (mplex, Yamux)
+	Muxer string
+	// No relay services, direct connections between peers only
+	NoRelay bool
+}
+
+type GeneralConfig struct {
```
Comment (sophoah) [inline internal/configs/bootnode/bootnode.go:131]: i think only NodeType and ShardID is really required here

File: internal/configs/bootnode/bootnode.go [hunk 13]
```diff
@@ -0,0 +1,316 @@
...
+	Port           int
+	AuthPort       int
+	RosettaEnabled bool
+	RosettaPort    int
+	ReadTimeout    string
+	WriteTimeout   string
+	IdleTimeout    string
+}
+
+type WsConfig struct {
```
Comment (sophoah) [inline internal/configs/bootnode/bootnode.go:243]: i don't think authport is required

File: internal/configs/bootnode/bootnode.go [hunk 14]
```diff
@@ -0,0 +1,316 @@
...
+	case PriceLimit:
+		*s = v
+	default:
+		return fmt.Errorf("PriceLimit.UnmarshalTOML: %T", data)
+	}
+	return nil
+}
+
+func (s PriceLimit) MarshalTOML() ([]byte, error) {
+	if s > 1_000_000_000 {
```
Comment (sophoah) [inline internal/configs/bootnode/bootnode.go:312]: do we need that for bootnode config ?

File: internal/configs/bootnode/bootnode.go [hunk 15]
```diff
@@ -0,0 +1,316 @@
...
+		StakingRPCsEnabled: bnc.RPCOpt.StakingRPCsEnabled,
+		LegacyRPCsEnabled:  bnc.RPCOpt.LegacyRPCsEnabled,
+		RpcFilterFile:      bnc.RPCOpt.RpcFilterFile,
+		RateLimiterEnabled: bnc.RPCOpt.RateLimterEnabled,
+		RequestsPerSecond:  bnc.RPCOpt.RequestsPerSecond,
+		EvmCallTimeout:     evmCallTimeout,
+	}
+}
+
+type DnsSync struct {
```
Comment (sophoah) [inline internal/configs/bootnode/bootnode.go:89]: Do we need in the bootnode config ?

File: internal/configs/bootnode/bootnode.go (hunk 13 above)
Comment (sophoah) [reply internal/configs/bootnode/bootnode.go:243]: and no auth port

File: internal/configs/bootnode/bootnode.go (hunk 5 above)
Comment (GheisMohammadi) [reply internal/configs/bootnode/bootnode.go:280]: removed

File: internal/configs/bootnode/bootnode.go (hunk 6 above)
Comment (GheisMohammadi) [reply internal/configs/bootnode/bootnode.go]: removed

File: internal/configs/bootnode/bootnode.go (hunk 7 above)
Comment (GheisMohammadi) [reply internal/configs/bootnode/bootnode.go]: removed

File: internal/configs/bootnode/bootnode.go (hunk 9 above)
Comment (GheisMohammadi) [reply internal/configs/bootnode/bootnode.go]: Removed

File: internal/configs/bootnode/bootnode.go (hunk 10 above)
Comment (GheisMohammadi) [reply internal/configs/bootnode/bootnode.go]: Removed

File: internal/configs/bootnode/bootnode.go (hunk 11 above)
Comment (GheisMohammadi) [reply internal/configs/bootnode/bootnode.go]: Removed

File: internal/configs/bootnode/bootnode.go (hunk 13 above)
Comment (GheisMohammadi) [reply internal/configs/bootnode/bootnode.go:166]: removed

File: internal/configs/bootnode/bootnode.go (hunk 14 above)
Comment (GheisMohammadi) [reply internal/configs/bootnode/bootnode.go]: Removed

File: internal/configs/bootnode/bootnode.go (hunk 15 above)
Comment (GheisMohammadi) [reply internal/configs/bootnode/bootnode.go]: Removed

File: internal/configs/bootnode/config.go [hunk 16]
```diff
@@ -0,0 +1,252 @@
...
+			key.SerializeToHexStr())
+	}
+	epoch := conf.networkType.ChainConfig().StakingEpoch
+	numShards := conf.shardingSchedule.InstanceForEpoch(epoch).NumShards()
+	shardID := new(big.Int).Mod(pubKey.Big(), big.NewInt(int64(numShards)))
+	return uint32(shardID.Uint64()), nil
+}
+
+// ValidateConsensusKeysForSameShard checks if all consensus public keys belong to the same shard
+func (conf *ConfigType) ValidateConsensusKeysForSameShard(pubkeys multibls.PublicKeys, sID uint32) error {
```
Comment (sophoah) [inline internal/configs/bootnode/config.go:232]: i don't think this is used by the bootnode right ?

File: internal/configs/bootnode/config.go [hunk 17]
```diff
@@ -0,0 +1,252 @@
...
+// SetShardingSchedule sets the sharding schedule for all config instances.
+func SetShardingSchedule(schedule shardingconfig.Schedule) {
+	defaultConfig.SetShardingSchedule(schedule)
+	for _, config := range shardConfigs {
+		config.SetShardingSchedule(schedule)
+	}
+}
+
+// ShardIDFromKey returns the shard ID statically determined from the input key
+func (conf *ConfigType) ShardIDFromKey(key *bls_core.PublicKey) (uint32, error) {
```
Comment (sophoah) [inline internal/configs/bootnode/config.go:218]: i don't think this is used by the bootnode right ?

File: internal/configs/bootnode/config.go [hunk 18]
```diff
@@ -0,0 +1,252 @@
...
+	return peerID
+}
+
+// ShardingSchedule returns the sharding schedule for this node config.
+func (conf *ConfigType) ShardingSchedule() shardingconfig.Schedule {
+	return conf.shardingSchedule
+}
+
+// SetShardingSchedule sets the sharding schedule for this node config.
+func (conf *ConfigType) SetShardingSchedule(schedule shardingconfig.Schedule) {
```
Comment (sophoah) [inline internal/configs/bootnode/config.go:205]: i don't think this is used by the bootnode right ?

File: internal/configs/bootnode/bootnode.go (hunk 8 above)
Comment (GheisMohammadi) [reply internal/configs/bootnode/bootnode.go:154]: removed

File: internal/configs/bootnode/bootnode.go (hunk 12 above)
Comment (GheisMohammadi) [reply internal/configs/bootnode/bootnode.go:100]: removed

File: internal/configs/bootnode/config.go (hunk 16 above)
Comment (GheisMohammadi) [reply internal/configs/bootnode/config.go:232]: removed

File: internal/configs/bootnode/config.go (hunk 17 above)
Comment (GheisMohammadi) [reply internal/configs/bootnode/config.go:218]: removed

File: internal/configs/bootnode/config.go (hunk 18 above)
Comment (GheisMohammadi) [reply internal/configs/bootnode/config.go:205]: removed

File: internal/configs/bootnode/config.go [hunk 19]
```diff
@@ -0,0 +1,252 @@
...
+	conf.shardingSchedule = schedule
+}
+
+// SetShardingSchedule sets the sharding schedule for all config instances.
+func SetShardingSchedule(schedule shardingconfig.Schedule) {
```
[the remark this answers is not in the document]
Comment (GheisMohammadi) [reply internal/configs/bootnode/config.go:210]: removed

**Comment by sophoah** [comment]
Thanks @GheisMohammadi for all the removal. I am now looking the results returned by 3 APIs.

```bash
curl -d '{
  "jsonrpc":"2.0",
  "method":"hmyboot_getNodeMetadata",
  "params":[],
  "id":1
 }' -H 'Content-Type:application/json' -X POST '0.0.0.0:8888'
 ```
 returns 
 ```json
 {"jsonrpc":"2.0","id":1,"result":{"blskey":null,"chain-config":{"AllowlistEpoch":null,"FeeCollectEpoch":null,"chain-id":null,"eth-compatible-chain-id":null,"eth-compatible-shard-0-chain-id":null},"consensus":{"blocknum":0,"finality":0,"mode":"","phase":"","viewChangeId":0,"viewId":0},"current-block-number":0,"current-epoch":0,"dns-zone":"","is-archival":false,"is-backup":false,"is-leader":false,"network":"","node-unix-start-time":1725851267,"p2p-connectivity":{"connected":24,"not-connected":1,"total-known-peers":25},"peerid":"","role":"","shard-id":0,"version":"Harmony (C) 2023. bootnode, version v8448-v2024.2.0-52-g6d3b3fa7c (soph@ 2024-09-09T10:07:28+0700)"}}
```
I don’t believe the fields  `blskey`, `consensus`, `current-block-number`, `current-epoch`, `is-archival`, `is-backup`, `is-leader`, `shard-id` provide much value here. What are your thoughts? Additionally, the fields `network`, `peer-id`, `role` are not returning any results either.

```bash
curl -d '{
  "jsonrpc":"2.0",
  "method":"hmyboot_getPeerInfo",
  "params":[],
  "id":1
 }' -H 'Content-Type:application/json' -X POST '0.0.0.0:8888'
```
returns
 ```json
 {"jsonrpc":"2.0","id":1,"result":{"blocked-peers":[],"connected-peers":[],"peerid":""}}
```
Is it normal for the `connected-peers` field to be empty? Similarly, the `peerid` field also is not working.

`hmyboot_protocolVersion` looks ok.

**Comment by GheisMohammadi** [comment]
> ```shell
> hmyboot_getNodeMetadata
> ```

You're right. The current structure uses the Harmony node metadata format, which includes fields that are not particularly relevant to boot nodes. Since the boot node only utilizes specific fields related to its functionality, I agree it would make more sense to separate the boot node metadata from the Harmony node metadata. I will go ahead and refactor this to create a more tailored structure for boot nodes. Let me take care of that now.

**Comment by sophoah** [comment]
@GheisMohammadi really nice, here is the latest test result :
**hmyboot_getNodeMetadata**:
```bash
curl -d '{
  "jsonrpc":"2.0",
  "method":"hmyboot_getNodeMetadata",
  "params":[],
  "id":1
 }' -H 'Content-Type:application/json' -X POST '0.0.0.0:8888'
 ```
 ```json
 {"jsonrpc":"2.0","id":1,"result":{"network":"localnet","node-unix-start-time":1725940221,"p2p-connectivity":{"connected":24,"not-connected":1,"total-known-peers":25},"peerid":"Qmc1V6W7BwX8Ugb42Ti8RnXF1rY5PF7nnZ6bKBryCgi6cv","version":"Harmony (C) 2023. bootnode, version v8452-v2024.2.0-56-g8e7a164d5 (soph@ 2024-09-10T10:50:07+0700)"}}
 ```

 ---

 **hmyboot_getPeerInfo**:
 ```bash
curl -d '{
  "jsonrpc":"2.0",
  "method":"hmyboot_getPeerInfo",
  "params":[],
  "id":1
 }' -H 'Content-Type:application/json' -X POST '0.0.0.0:8888'
```
```json
{"jsonrpc":"2.0","id":1,"result":{"c":{"connected":24,"not-connected":1,"total-known-peers":25},"connected-peers":["QmTnxxzCBD22quNagCFoDqvH6tQrMBAKqjNzx6Hbhog9ms","QmX48ysgKnWBNrXKpFQ1nFVAynCqYBWYjJ4QaHwkf4g4Y5","QmVyGzpU2muKMQ2KXG2WTrKBJazZ77WZmG4dqdpsaTzy8K","QmeLsMmiNdbzPiD8iRenkVKxXronhCxUQ6KAoxmjWqBNcU","QmP8tBd8KGEbB3xcW3ajFfLF4y8HVXpvdRwRDptuAQbJ74","QmQDCFekk8PuomyPSdAuiF4GCUBaCJFw6Uz3k35CrXAMRD","QmUBXaTCyH9fpzqLmpvgkbT3FvUqywJbfNNWoiKnrL2Cb5","QmXRi4nKeyNaXeiVR3MDwBbkZCvDAKavkyMD7aoPwUBw2Q","QmPsY3ndVWrhHW3M3nSVEocmCZtxHJm3NjXUpiHmX5wAph","QmeY1spswbWY7yMBCEc367aWZnH12zS2MWQgYQcLYEQFQJ","QmTwG4pCtE1gaY7gm5nZg1HtHK2AR2oL1HMGXPucDDFHWt","QmUJLE7wytFaKHRP25run8xRbGdHqTvyeFrDHLywzFkNGY","QmV7oduzPb1urLyaCSC942VHDLT4w6QMCZgXCWLxJcduWf","QmXJSA31NzmzKWmjGatsBCkBCFoYWHgeKV3E2cbeXqQyGe","QmZ17k6iWv2Dj4RHHhdX1VZ8H5UKze4Yue3RNRRS9skTKq","QmdhkDwWbgiqu6V7U3eD5j95Cvjf54NwgaHv9w6vBnVbGY","QmQ82zJZrtEANUkp2DJDg9VGixbSn6oDL9AD1mKPMY74yG","QmTgEFsc7tVUJ4CnCgmkuG5ANm96SEHrNkgJgA1PCw2maz","QmcCxKtPuLhTenq6J4DoFKEeEcRZJW74e9SvhFkGibThPr","QmWnNpwnUJf2mhtwkr7AZbMxF4CDHFfoe7AaPbX7zF3J7a","QmYYDjzMgBnxQrYUrqTAYcSFpJ2rT6XCZjnFreYBwSdiuZ","QmZLQ45S2aEnJohFuuTc37x7LkMgTUUGN8zFjJ7K7wqFAy","QmXhm56KuvZ9idXERpp12gmJAmESxb6nYki1zEqGRzckfv","QmdGkuubG1wHgMEsLXEhm2jPDj3s2pZ18tKZ5T94F2gqtb"],"known-peers":["QmUBXaTCyH9fpzqLmpvgkbT3FvUqywJbfNNWoiKnrL2Cb5","QmWnNpwnUJf2mhtwkr7AZbMxF4CDHFfoe7AaPbX7zF3J7a","QmZ17k6iWv2Dj4RHHhdX1VZ8H5UKze4Yue3RNRRS9skTKq","QmdhkDwWbgiqu6V7U3eD5j95Cvjf54NwgaHv9w6vBnVbGY","QmTgEFsc7tVUJ4CnCgmkuG5ANm96SEHrNkgJgA1PCw2maz","QmeLsMmiNdbzPiD8iRenkVKxXronhCxUQ6KAoxmjWqBNcU","QmXRi4nKeyNaXeiVR3MDwBbkZCvDAKavkyMD7aoPwUBw2Q","QmTnxxzCBD22quNagCFoDqvH6tQrMBAKqjNzx6Hbhog9ms","QmVyGzpU2muKMQ2KXG2WTrKBJazZ77WZmG4dqdpsaTzy8K","QmQ82zJZrtEANUkp2DJDg9VGixbSn6oDL9AD1mKPMY74yG","QmdGkuubG1wHgMEsLXEhm2jPDj3s2pZ18tKZ5T94F2gqtb","QmXJSA31NzmzKWmjGatsBCkBCFoYWHgeKV3E2cbeXqQyGe","QmcCxKtPuLhTenq6J4DoFKEeEcRZJW74e9SvhFkGibThPr","QmYYDjzMgBnxQrYUrqTAYcSFpJ2rT6XCZjnFreYBwSdiuZ","QmTwG4pCtE1gaY7gm5nZg1HtHK2AR2oL1HMGXPucDDFHWt","QmeY1spswbWY7yMBCEc367aWZnH12zS2MWQgYQcLYEQFQJ","Qmc1V6W7BwX8Ugb42Ti8RnXF1rY5PF7nnZ6bKBryCgi6cv","QmZLQ45S2aEnJohFuuTc37x7LkMgTUUGN8zFjJ7K7wqFAy","QmV7oduzPb1urLyaCSC942VHDLT4w6QMCZgXCWLxJcduWf","QmXhm56KuvZ9idXERpp12gmJAmESxb6nYki1zEqGRzckfv","QmPsY3ndVWrhHW3M3nSVEocmCZtxHJm3NjXUpiHmX5wAph","QmX48ysgKnWBNrXKpFQ1nFVAynCqYBWYjJ4QaHwkf4g4Y5","QmP8tBd8KGEbB3xcW3ajFfLF4y8HVXpvdRwRDptuAQbJ74","QmQDCFekk8PuomyPSdAuiF4GCUBaCJFw6Uz3k35CrXAMRD","QmUJLE7wytFaKHRP25run8xRbGdHqTvyeFrDHLywzFkNGY"],"peerid":"Qmc1V6W7BwX8Ugb42Ti8RnXF1rY5PF7nnZ6bKBryCgi6cv"}}
```

---

**hmyboot_protocolVersion**:
```bash
curl -d '{
  "jsonrpc":"2.0",
  "method":"hmyboot_protocolVersion",
  "params":[],
  "id":1
 }' -H 'Content-Type:application/json' -X POST '0.0.0.0:8888'
```
```json
{"jsonrpc":"2.0","id":1,"result":"0x1"}
```

**Comment by sophoah** [comment]
@mur-me let's update our devops to add the network flag accordingly when we deploy that PR to our bootnode `-network`

**Comment by sophoah** [comment]
Note that the flag is just informative/cosmetic, it doesn't play any role in the bootnode logic behavior here.

**Comment by GheisMohammadi** [comment]
Thanks for the review and great comments

**Review by sophoah — APPROVED** [review]

**Pull request merged**